### PR TITLE
Change findtag to work as a filter

### PR DESCRIPTION
**Pull request by glennljw**
This PR changes the `findtag` command to work as a filter. 

**Each** `findtag` command activates a tag to be searched.
e.g. 
1. `findtag friends` adds `friends` as a tag to be searched for
1. `findtag colleagues` adds `colleagues` to pre-existing search, now containing both `colleagues` and `friends`

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/FindTagCommand.java [hunk 1]
```diff
@@ -25,33 +28,49 @@
...
+            model.addActivatedTag(tag);
         }
 
-        for (String matched : matchedKeywords) {
-            Tag currTag = new Tag(matched);
-            model.addActivatedTag(currTag);
+        if (keywords.size() == 0) {
+            throw new CommandException(MESSAGE_USAGE);
         }
+        ObservableList<Tag> activatedTagList = model.getActivatedTagList();
```
Comment (weijuey) [inline src/main/java/seedu/address/logic/commands/FindTagCommand.java:68]: Did you need this variable? 🤔

File: src/test/java/seedu/address/logic/commands/FindTagCommandTest.java [hunk 2]
```diff
@@ -31,14 +40,19 @@ void equals() {
         TagContainsKeywordsPredicate secondPredicate =
                 new TagContainsKeywordsPredicate(Collections.singletonList("second"));
 
-        FindTagCommand findFirstCommand = new FindTagCommand(firstPredicate);
-        FindTagCommand findSecondCommand = new FindTagCommand(secondPredicate);
```
Comment (weijuey) [inline src/test/java/seedu/address/logic/commands/FindTagCommandTest.java:35]: Looks like you can also delete the predicates above 😄

File: src/test/java/seedu/address/logic/commands/FindTagCommandTest.java [hunk 3]
```diff
@@ -52,25 +66,57 @@ void equals() {
...
     }
 
     @Test
-    public void execute_singleKeyword_multiplePersonsFound() {
+    public void execute_zeroKeywordsNoActivatedTags_throwsCommandException() {
+        String expectedMessage = FindTagCommand.MESSAGE_USAGE;
+        List<String> emptyKeywords = new ArrayList<>();
+        FindTagCommand command = new FindTagCommand(emptyKeywords);
+        TagContainsKeywordsPredicate predicate = new TagContainsKeywordsPredicate(emptyKeywords);
+        expectedModel.updateFilteredPersonList(predicate);
```
Comment (weijuey) [inline src/test/java/seedu/address/logic/commands/FindTagCommandTest.java:88]: I think you don't need to have this line here!

**Review by weijuey — CHANGES_REQUESTED** [review]
Small nits, but otherwise lgtm! Good work squeezing some tests in too 😅

File: src/test/java/seedu/address/logic/commands/FindTagCommandTest.java (hunk 2 above)
Comment (glennljw) [reply src/test/java/seedu/address/logic/commands/FindTagCommandTest.java:35]: Thanks for spotting this! 😄

File: src/test/java/seedu/address/logic/commands/FindTagCommandTest.java (hunk 3 above)
Comment (glennljw) [reply src/test/java/seedu/address/logic/commands/FindTagCommandTest.java:88]: Yup you're right! Removed them!

File: src/main/java/seedu/address/logic/commands/FindTagCommand.java (hunk 1 above)
Comment (weijuey) [reply src/main/java/seedu/address/logic/commands/FindTagCommand.java:68]: Oops, I didn't see it just below 🤣 my b.

**Review by weijuey — APPROVED** [review]

**Pull request merged**